specs/logging_change.adoc

```diff
@@ -0,0 +1,152 @@
+After implementation of new logging framework there will be requirement of changes in all the modules of Tendrl where logging is used.
+
+#### Use Cases
+
+Logs and operation logs need be treated as messages.
+
+#### Proposed change
+
+* *Importing logging and calling logger functions* will be replaced by:
+** Importing Message and Event class from commons
+** Introduce a try-except block and do :
+
+*** Create a Message object by passing the required arguments to it and then
+ pass this message object to the event class
+*** Catch the exception if the event instantiation fails.
+** There will be 3 arguments that will be mandatory to create the Message
+object:
+ *** priority
+ *** publisher
+ **** eg. CEPH-INTEGRATION, GLUSTER-INTEGRATION, NODE-AGENT, COMMONS, ALERTING, PERFORMANCE-MONITORING
+ *** payload   <Json>
+ **** If the message is for a normal log, then the payload should look like :
+
+ {
+    "message" : "User Message"
+ }
+
+ **** If the message is for operation updates, then mandatory parameters for payload are *flow_id and job_id*
+
+ {
+    "flow_id" : "Flow ID",
+    "job_id" : "Job_ID",
+    "message" :"User Message",
+    .
+    .Optional fields
+    .
+ }
+
+#### Alternatives
+
+None
+
+#### Data model impact:
+
+None
+
+#### Impacted Modules:
+
+#### Tendrl/common impact
+
+All the logging instances to be replaced by the new framework as mentioned in proposed change.
+
+#### Tendrl/node_agent impact
+
+All the logging instances to be replaced by the new framework as mentioned in proposed change.
+
+#### Tendrl/gluster_integration impact
+
+All the logging instances to be replaced by the new framework as mentioned in proposed change.
+
+#### Tendrl/ceph_integration impact
+
+All the logging instances to be replaced by the new framework as mentioned in proposed change.
+
+#### Tendrl/alerting impact
+
+All the logging instances to be replaced by the new framework as mentioned in proposed change.
+
+#### Tendrl/performance_monitoring impact
+
+All the logging instances to be replaced by the new framework as mentioned in proposed change.
+
+#### Security impact:
+None
+
+#### Notifications/Monitoring impact
+
+None
+
+#### Other end user impact:
+
+None
+
+#### Performance impact:
+
+None
+
+#### Other deployer impact:
+
+None
+
+#### Developer impact:
+
+None
+
+#### Implementation:
+* Import Message and Event class from commons
+* Create a message object.
+* Pass the message object as an argument to make an Event Object.
+----
+from tendrl.commons.message import Message
+import tendrl.commons.event
+try:
+    payload={   "flow_id" : "Flow ID ",
+                "job_id"  : "Job ID",
+                "message" : "User message"
+            }
+    event.Event(Message(Message.priorities.PRIORITY,
+    Message.publishers.PUBLISHER,payload))
+except event.EventFailed:
+    print(traceback.format_exc())
```

Comment:
@r0h4n no we can't log because if the socket fails how we can log, this exception is for socket fail, it just a print in consloe